wgsl/analyze/Grammar.py

```diff
@@ -1636,9 +1640,14 @@ def pretty_str(rule):
             if rule.is_symbol_name():
                 return rule.content
             if isinstance(rule,Choice):
-                return " | ".join([pretty_str(i) for i in rule])
+                inside = " | ".join([pretty_str(i) for i in rule])
```

Comment:
Can you join these with a newline? so you'd get

```
 assignment_statement:
   ( lhs_expression ( equal | compound_assignment_operator ) expression
   | underscore equal expression )
```

instead of: 

```
 assignment_statement: ( lhs_expression ( equal | compound_assignment_operator ) expression | underscore equal expression )
```

Reply:
Can do.

Reply:
Posted an update that does it.

But then with nesting you get stuff like this:


```
argument_expression_list: paren_left (
   (
   ( expression comma )+
 | ε
) expression (
   comma
 | ε
)
 | ε
) paren_right
```

Reply:
which still reads ok.  But it's handy to have a parens-matching editor like vim.

Reply:
But this is noticeably nicer:

```
attribute: (
   attr 'align' paren_left expression attrib_end
 | attr 'binding' paren_left expression attrib_end
 | attr 'builtin' paren_left builtin_value_name attrib_end
 | attr 'const'
 | attr 'group' paren_left expression attrib_end
 | attr 'id' paren_left expression attrib_end
 | attr 'interpolate' paren_left interpolation_type_name attrib_end
 | attr 'interpolate' paren_left interpolation_type_name comma interpolation_sample_name attrib_end
 | attr 'invariant'
 | attr 'location' paren_left expression attrib_end
 | attr 'size' paren_left expression attrib_end
 | attr 'workgroup_size' paren_left expression attrib_end
 | attr 'workgroup_size' paren_left expression comma expression attrib_end
 | attr 'workgroup_size' paren_left expression comma expression comma expression attrib_end
 | attr 'vertex'
 | attr 'fragment'
 | attr 'compute'
)

```